web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -120,6 +108,145 @@
         };
 
         return {
+          /**
+           * @ngdoc method
+           * @methodOf gn_map.service:gnMap
+           * @name gnMap#createLayerFromNotation
+           *
+           * @description
+           * Creates an ol.layer based on a string notation.
+           * Layer notation is defined like so:
+           * `{type=TYPE,arg1=VALUE,arg2=VALUE,...}`
+           * Handled TYPE values are:
+           *  * `osm`: OSM, no arg required
+           *  * `bing_aerial`: Bing Aerial background, required arg: `key`
+           *  * `stamen`: Stamen layers, required arg: `name`
+           *  * `wms`: generic WMS layer, required args: `name`, `url`
+           *  * `wmts`: generic WMTS layer, required args: `name`, `url`
+           *  * `tms`: generic TMS layer, required arg: `url`
+           * This will return a promise with the layer in it
+           *
+           * @param {string} notation normalized layer description
+           * @param {string} title optional title
+           * @param {ol.Map} map required for WMTS and WMS
+           * @return {Promise} promise with the layer as result
+           */
+          createLayerFromNotation: function(notation, title, map) {
+            var testRE = /{type=[^,]+,?(?:[^,]+=[^,]+,?)*}/;
```

Comment:
The format should not be a string of a fake JSON object, please use an object instead.